indico_toolkit/indico_wrapper/modelop.py

```diff
@@ -0,0 +1,200 @@
+            GraphQLRequest(
+                """
+                mutation retrainMoonbowModelGroup($modelGroupId: Int!) {
+                    retrainModelGroup(modelGroupId: $modelGroupId, forceRetrain: true) {
+                        id
+                        status
+                    }
+                }    
+                """,
+                {"modelGroupId": model_group_id},
+            )
+        )
+
+    def update_model_settings(
+        self, model_group_id: int, model_type: str, **kwargs
+    ) -> dict[str, object]:
+        """
+        Update group model settings based on specified model training options.
+
+        Args:
+             model_group_id (int): id of model group
+             model_type (str): type of model ("text_extraction", "text_classification")
+             **kwargs: Advanced Model Training Options
+
+            Default values of Advanced Model Training Options and the avaiable parameters:
+                    For Text Extraction Model:
+                        max_empty_chunk_ratio : 1.0 (min of 0, no max value: a large number effectivly turns this option off)
+                        auto_negative_scaling : True
+                        optimize_for : "predict_speed" ( "predict_speed", "accuracy", "speed", "accuracy_fp16" and "predict_speed_fp16")
+                        subtoken_predictions : True
+                        base_model : "roberta" ("roberta", "small" (distilled version of RoBERTa), "multilingual", "fast", "textcnn", "fasttextcnn")
+                        class_weight : "sqrt" ("linear", "sqrt", "log", None)
+
+                    For Text Classification Model:
+                        model_type : "standard" (“tfidf_lr”, “tfidf_gbt”, “standard”, “finetune”)
+        Returns:
+            dict: Dictionary of advanced model training options
+        """
+        model = self.client.call(
+            GraphQLRequest(
+                """
+                mutation updateModelGroup($modelGroupId: Int!, $modelTrainingOptions: JSONString) {
+                    updateModelGroupSettings(
+                      modelGroupId: $modelGroupId
+                      modelTrainingOptions: $modelTrainingOptions
+                  ) {
+                      modelOptions {
+                          id
+                          modelTrainingOptions
+                        }
+                    }
+                }
+                """,
+                {
+                    "modelGroupId": model_group_id,
+                    "modelTrainingOptions": json.dumps(
+                        self._parameter_check(model_type, **kwargs)
+                    ),
+                },
+            )
+        )
+        options = json.loads(
+            model["updateModelGroupSettings"]["modelOptions"]["modelTrainingOptions"]
+        )
+        options["id"] = model["updateModelGroupSettings"]["modelOptions"]["id"]
+        return options
+
+    def _parameter_check(self, model_type, **kwargs):
```

Comment:
This method does a bit much for my liking: both defining the valid options for two types of models as well as doing the validation itself. I'd lean towards using a data validation library like Pydantic, but what you have here works and doesn't require a new dependency. At the very least I think I'd move the valid options to the top of the module as constants to focus this method on just the validation.